src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,107 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.service.RegistrationServiceImpl.USER_MIN_ACCESSIBLE_AGE;
+import static core.basesyntax.service.RegistrationServiceImpl.WHITE_SPACE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.exceptions.UserRegistrationException;
+import core.basesyntax.model.User;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static final int USER_AGE_LESS_18_NOT_OK = 17;
+    private static final int USER_AGE_MORE_THAN_18_OK = 30;
+    private static final String USER_LOGIN_LENGTH_LESS_6_SYMBOLS_NOT_OK = "Bob";
+    private static final String USER_LOGIN_LENGTH_6_SYMBOLS_OK = "Bob123";
+    private static final String USER_PASSWORD_LENGTH_LESS_6_SYMBOLS_NOT_OK = "bob";
+    private static final String USER_PASSWORD_LENGTH_6_SYMBOLS_OK = "bob123";
```

Comment:
Try to test for user = null, 2 valid different users, and 2 valid similar users.

Reply:
Adding 2 valid different users is not necessary because if we cover all possible cases during other test. To my mind it will be valid if we add 2 users too.